zulip_bots/zulip_bots/bots/merels/libraries/database.py

```diff
@@ -0,0 +1,149 @@
+"""This module is used for managing, storing, and operating certain
```

Comment:
Recently, we added a feature to Zulip that allows bots to store data on a Zulip server permanently. You can read about it in our bot documentation, and `virtual_fs` and `tictactoe` are two bots using such a storage system. Would it be feasible for you to refactor your PR to use the bot storage system instead of sqlite?

Reply:
Unfortunately it's not really feasible for me to do that:

* [When I read performance considerations on this link](https://www.zulipchat.com/api/writing-bots). this means that I have to refactor every function that opens and closes the database connection, which I think is many
* Since it's still experimental, I didn't really want to touch on feature for now to maintain stability
* I store many columns on a record, so it's a little bit messier to work with if I switch to the feature as it only stores key and value 
* test_database.py will have to be rewritten somehow to *integrate* zulip storage system, which I don't know how to do

Reply:
I decided to port it anyway. Should be fine now.